roles/variables/defaults/main.yml

```diff
@@ -316,10 +317,10 @@ zookeeper_ssl_enabled: "{{ssl_enabled}}"
 zookeeper_ssl_mutual_auth_enabled: "{{ssl_mutual_auth_enabled}}"
 
 ### Deprecated- SASL Mechanism for Zookeeper Server to Server and Server to Client Authentication. Options are none, kerberos, digest. Server to server auth only working for digest-md5
-zookeeper_sasl_protocol: "{{sasl_protocol if sasl_protocol == 'kerberos' else 'none'}}"
+zookeeper_sasl_protocol: "{{ 'kerberos' if 'kerberos' in (sasl_protocol | confluent.platform.split_to_list) else 'none' }}"
 
 ### Authentication to put on ZK Server to Server connections. Available options: [mtls, digest, digest_over_tls].
-zookeeper_quorum_authentication_type: "{% if zookeeper_ssl_enabled and zookeeper_ssl_mutual_auth_enabled %}mtls{% elif zookeeper_sasl_protocol == 'digest' %}digest{% else %}none{% endif %}"
+zookeeper_quorum_authentication_type: "{% if zookeeper_ssl_enabled and zookeeper_ssl_mutual_auth_enabled %}mtls{% elif 'digest' in (sasl_protocol | confluent.platform.split_to_list) %}digest{% else %}none{% endif %}"
```

Comment:
digest is not an accepted value in sasl protocol. so this line shouldnt have been changed

Reply:
fixed it